base/regex.jl

```diff
@@ -269,7 +266,7 @@ function matchall(re::Regex, str::String, overlap::Bool=false)
             end
         end
 
-        push!(matches, SubString(str, ovec[1]+1, ovec[2]))
+        push!(matches, SubString(str, ovec[1]+1, prevind(str, ovec[2]+1)))
```

Comment:
Can you add a comment to explain what this does? It's quite hard to guess.

Reply:
You have to correct for the fact that ovec is 0-based and that ovec[2] is pointing at the last byte of the match. I am adding an appropriate comment to the source code

Reply:
You've added the comment above, but not here.

Reply:
This issue is in three places in this file. Each follows exactly the same rule. I thought to add it to the first occurrence from top of the file.
Do you think it is worth to add it everywhere?

Reply:
Mmm, tough call, but given how subtle this is, I'd rather be too verbose than not enough.

Reply:
ok - adding additional comment as it will not hurt